### PR TITLE
Add a way for sponsors to pay without asking for logos etc.

### PR DESCRIPTION
**Pull request by ericholscher**
This also adds the ability to see reports on specific promos.
Also gives a place to see how they display and test them before shipping them live.

### CODE REVIEW

File: readthedocs/donate/templates/donate/ethicalads.html [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+{% endblock %}
+
+
+{% block content %}
+  <h2>Pay for your Sponsorship</h2>
+
+  <p>
+    This form can be used to pay for your sponsorship of Read the Docs.
+    Thanks for helping make Open Source more sustainable!
+  </p>
```
Comment (agjohnson) [inline readthedocs/donate/templates/donate/ethicalads.html]: Same with the title and this paragraph block. Framing this as sponsorship is misleading.

File: readthedocs/donate/templates/donate/ethicalads.html [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+              value="50"
+              id="id_dollars_input"
+              style="display: none;" />
+          <select data-bind="value: dollars_select, visible: dollars_select() != 'custom'">
+            <option value="custom">{% trans "Custom amount" %}</option>
+            <option value="2000">$2,000</option>
+            <option value="3000">$3,000</option>
+            <option value="5000" selected>$5,000</option>
+            <option value="10000">$10,000</option>
+          </select>
```
Comment (agjohnson) [inline readthedocs/donate/templates/donate/ethicalads.html]: This select shouldn't exist at all.

File: readthedocs/donate/views.py [hunk 3]
```diff
@@ -15,12 +15,28 @@
...
-from .forms import SupporterForm
+from .forms import SupporterForm, EthicalAdForm
 from .mixins import DonateProgressMixin
 
 log = logging.getLogger(__name__)
 
 
+class PayAdsView(StripeMixin, CreateView):
+
+    """Create a donation locally and in Stripe"""
```
Comment (agjohnson) [inline readthedocs/donate/views.py]: s/donation/advertising payment/

File: readthedocs/donate/templates/donate/ethicalads.html (hunk 2 above)
Comment (ericholscher) [inline readthedocs/donate/templates/donate/ethicalads.html]: I left this to set the standard expectation that they should be paying us $5k test buys. Seemed simpler to have this select than rewrite how we were handling the actual data.

Comment (agjohnson) [inline readthedocs/donate/templates/donate/ethicalads.html]: It does seem a bit odd still though. Altering the form should be straight forward. You'd need to remove the `<select>`, drop the `visible` data bind parameter on `<input>`, and drop the `style` parameter on the `<input>` as well. This should give a permanent text input.

File: readthedocs/donate/templates/donate/promo_detail.html [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+</h3>
+
+<div class="example" style="width: 30%;">
+<a href="{{ promo.link }}"><img width=120 height=90 src="{{ promo.image }}"></a>
+<br>
+{{ promo.text }}
+
+</div>
+
+<br>
```
Comment (agjohnson) [inline readthedocs/donate/templates/donate/promo_detail.html]: Whitespace shouldn't be added with html elements, this should be a css change.

File: readthedocs/donate/templates/donate/promo_detail.html [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+<table>
+  <tr>
+    <th><strong>Day (UTC)</strong></th>
+    <th><strong>Views</strong></th> 
+    <th><strong>Clicks</strong></th>
+    <th><strong>CTR</strong></th>
```
Comment (agjohnson) [inline readthedocs/donate/templates/donate/promo_detail.html]: The `<strong>` here should also be replace be css rules. We shouldn't dictate styling in the HTML

File: readthedocs/donate/templates/donate/promo_detail.html [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    <td><code>{{ day.views }}</code></td> 
+    <td><code>{{ day.clicks }}</code></td>
+    <td><code>{{ day.click_ratio }}%</code></td>
+  </tr>
+  {% endfor %}
+  <tr>
+    <td><strong>Total (over all time)</strong> </td>
+    <td><strong>{{ promo.total_views }}</strong></td> 
+    <td><strong>{{ promo.total_clicks }}</strong></td>
+    <td><strong>{{ promo.total_click_ratio }}%</strong></td>
```
Comment (agjohnson) [inline readthedocs/donate/templates/donate/promo_detail.html]: Same for styling here. This should perhaps be a rule for `table > tr.table-totals` or somesuch.

File: readthedocs/donate/views.py [hunk 7]
```diff
@@ -77,6 +77,26 @@ def get_template_names(self):
         return [self.template_name]
 
 
+class PromoDetailView(TemplateView):
+    template_name = 'donate/promo_detail.html'
+
+    def get_context_data(self, **kwargs):
+        promo_slug = kwargs['promo_slug']
+        slugs = promo_slug.split(',')
+        days = int(self.request.GET.get('days', 90))
```
Comment (agjohnson) [inline readthedocs/donate/views.py]: Input isn't sanitized here, this should be wrapped for exception handling on `TypeError`

**Comment by designsb** [comment]
https://github.com/agjohnson
https://travis-ci.org
https://travis-ci.org/
https://github.com/rtfd/readthedocs.org/labels/PR%3A%20hotfix

> thank you

**Comment by ericholscher** [comment]
This has been live for quite a while -- can I just merge this @agjohnson, or do we care enough about the tidbits to fix it?

**Comment by agjohnson** [comment]
My review still stands, but feel free to open issues to track the clean up efforts if you aren't going to handle them. I think this module makes more sense being pulled out of RTD eventually anyways

**Comment by ericholscher** [comment]
Added #2691

**Pull request merged**